sycl/doc/extensions/supported/sycl_ext_intel_kernel_args_restrict.asciidoc

```diff
@@ -8,138 +9,128 @@
-      buf2_m[id]=id*2;
-    }
+  ...
+  [[intel::kernel_args_restrict]]
+  void operator()(item<1> item)
+  {
+    int id = item[0];
+    buf1_m[id]=id;
+    buf2_m[id]=id*2;
+  }
 };
 ----
 
 
 
-== Issues
-
-None.
+== Specification
 
-//. Title
-//+
-//--
-//*RESOLUTION*: Description
-//--
-
-== Feature test macro
+=== Feature test macro
 
 This extension provides a feature-test macro as described in the core SYCL
-specification section 6.3.3 "Feature test macros". Therefore, an implementation
-supporting this extension must predefine the macro `SYCL_EXT_INTEL_KERNEL_ARGS_RESTRICT`
-to one of the values defined in the table below. Applications can test for the
-existence of this macro to determine if the implementation supports this
-feature, or applications can test the macro�s value to determine which of the
-extension�s APIs the implementation supports.
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_KERNEL_ARGS_RESTRICT` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
 
 [%header,cols="1,5"]
 |===
-|Value |Description
-|1     |Initial extension version. Base features are supported.
+|Value
+|Description
+
+|1
+|Initial version of this extension.
 |===
 
-== Revision History
-
-[cols="5,15,15,70"]
-[grid="rows"]
-[options="header"]
-|========================================
-|Rev|Date|Author|Changes
-|1|2019-11-11|Michael Kinsner|*Initial public working draft*
-|========================================
-
-//************************************************************************
-//Other formatting suggestions:
-//
-//* Use *bold* text for host APIs, or [source] syntax highlighting.
-//* Use +mono+ text for device APIs, or [source] syntax highlighting.
-//* Use +mono+ text for extension names, types, or enum values.
-//* Use _italics_ for parameters.
-//************************************************************************
+
+=== Add new entry to Table 180 in section 5.8.1 (Kernel attributes)
+
+[width="100%",options="header",separator="@",cols="65%,35%"]
+|====
+@ SYCL attribute @ Description
+a@
+[source]
+----
+intel::kernel_args_restrict
+----
+   a@ Hint to the compiler equivalent to specifying the C99 `restrict`
+   attribute on all pointers and accessors (treated as if each accessor was a
+   pointer) that are captured as kernel arguments. This effect is equivalent to
+   annotating `restrict` on *all* kernel pointer arguments in an OpenCL or
+   SPIR-V kernel.
+
+The attribute forms an unchecked assertion, in that implementations do not need
+to check/confirm the pre-condition in any way.  If a user applies `intel::kernel_args_restrict`
+to a kernel, but there is in fact aliasing between kernel pointer arguments at runtime,
```

Comment:
```suggestion
to a kernel, but there is in fact aliasing between accessors at runtime,
```

Likewise, here we should talk about accessors, not internal pointers.

Reply:
We can't talk about just accessors because pointers can be captured as well.  Are you ok with "between accessors and/or pointers"?

Reply:
Yes, good point.  I like this new wording.